### PR TITLE
Fix/165089509 refactor profile form definition

### PR DESCRIPTION
**Pull request by diablourbano**
This refactor moves the FormDefinitions (Structures) from `NewProfile` and `ProfileForm` to `components/ProfileFormDef` to accomplish:

- reduce file size for `NewProfile` and `ProfileForm`
- reuse `InputField` component for `NewProfile`, `ProfileForm`
- reuse `FormStruct` definitions for `NewProfile`, `ProfileForm` and the upcoming `Create/Edit Persona`

### CODE REVIEW

File: src/components/ProfileRelatedForm/ProfileRelatedForm.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    keyboardType: locals.keyboardType || 'default',
+    style: {
+      fontSize: 24,
+      lineHeight: 0,
+    },
+    placeholder: config.placeholder || '',
+    ...config.inputProps,
+  };
+
+  const propsToInclude = {};
```
Comment (jsidorenko) [inline src/components/ProfileRelatedForm/ProfileRelatedForm.js]:
```suggestion
  const additionalProps = {};
```

File: src/screens/Profile/ProfileForm.js [hunk 2]
```diff
@@ -22,11 +22,16 @@ import React from 'react';
 import { Dimensions } from 'react-native';
 import t from 'tcomb-form-native';
 import styled from 'styled-components/native';
-import TextInput from 'components/TextInput';
 import { Wrapper } from 'components/Layout';
 import { spacing } from 'utils/variables';
 import Button from 'components/Button';
-import { isValidEmail, isValidName, isValidCityName } from 'utils/validators';
+import { InputTemplate, Form } from 'components/ProfileRelatedForm';
```
Comment (jsidorenko) [inline src/screens/Profile/ProfileForm.js]: What do you think about renaming `ProfileForm` to `EditProfile` and `components/ProfileRelatedForm` to `components/ProfileForm`?

**Comment by jsidorenko** [comment]
Despite the comments above everything looks good!

File: src/screens/Profile/ProfileForm.js (hunk 2 above)
Comment (diablourbano) [reply src/screens/Profile/ProfileForm.js]: I like it, thanks for that

File: src/components/ProfileRelatedForm/ProfileRelatedForm.js (hunk 1 above)
Comment (diablourbano) [reply src/components/ProfileRelatedForm/ProfileRelatedForm.js]: done

**Comment by diablourbano** [comment]
@jsidorenko it's ready, thanks for the suggestions

**Pull request merged**